R-package/R/lgb.train.R

```diff
@@ -142,6 +142,9 @@ lgb.train <- function(params = list(),
   }
   end_iteration <- begin_iteration + params[["num_iterations"]] - 1L
 
+  # Construct datasets, if needed
```

Comment:
```suggestion
  # Construct datasets, if needed
  data$update_params(params = params)
```

To avoid this error:

> [LightGBM] [Fatal] Reducing `min_data_in_leaf` with `feature_pre_filter=true` may cause unexpected behaviour for features that were pre-filtered by the larger `min_data_in_leaf`.You need to set `feature_pre_filter=false` to dynamically change the `min_data_in_leaf`.